types/transaction.go

```diff
@@ -61,4 +61,21 @@ type Header struct {
 	Note        []byte     `codec:"note"`
 	GenesisID   string     `codec:"gen"`
 	GenesisHash Digest     `codec:"gh"`
+
+	// Group specifies that this transaction is part of a
+	// transaction group (and, if so, specifies the hash
+	// of a TxGroup).
+	Group Digest `codec:"grp"`
+}
+
+// TxGroup describes a group of transactions that must appear
+// together in a specific order in a block.
+type TxGroup struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	// Transactions specifies a list of transactions that must appear
+	// together, sequentially, in a block in order for the group to be
+	// valid.  Each hash in the list is a hash of a transaction with
+	// the `Group` field omitted.
+	Transactions []Digest `codec:"txlist"`
```

Comment:
Although I believe it was copied from our main repo, I think that the name `TxGroup` is a bit misleading. I expected to see `Transaction []Transaction` but instead it contains the hashes as `Digest`. Maybe we should consider changing the name to `TxGroupHahses`? and the comment accordingly? @zeldovich - Was there a reason for name that I missing?

Reply:
I don't think there's anything deep going on here.  Renaming it seems fine.  I think consistency between the SDK and the `go-algorand` codebase is important, so that means renaming it in `go-algorand` too if you care enough about renaming.

Reply:
I'll rename here and there.